src/pages/index.js

```diff
@@ -31,37 +58,78 @@ const profileEditModal = new PopupWithForm(
 );
 profileEditModal.setEventListeners();
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
+const avatarModal = new PopupWithForm("#avatar-modal", handleAvatarSubmit);
+avatarModal.setEventListeners();
+
+const deleteModal = new PopupDelete("#delete-modal");
+deleteModal.setEventListeners();
 
-section.renderItems();
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+api.getUserInfo().then((userData) => {
+  userInfo.setAvatar(userData);
+  userInfo.setUserInfo({
+    name: userData.name,
+    about: userData.about,
+  });
+});
 
 // Functions
 
-function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick).getView();
-  return card;
+function renderLoading(isLoading, button) {
+  if (isLoading) {
+    button.textContent = "Saving...";
+  } else {
+    button.textContent = "Save";
+  }
 }
 
 // Event Handlers
 
 function handleProfileEditSubmit(inputValues) {
   console.log(inputValues);
-  userInfo.setUserInfo(inputValues);
+  renderLoading(true, editSubmitButton);
+  api
+    .updateUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+    })
+    .finally(() => {
+      renderLoading(false, editSubmitButton);
+    });
   profileEditModal.close();
   profileEditValidator.disableButton();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


And thus disabling should be in `then`


**Please, correct it everywhere**